.github/workflows/soundness.yml

```diff
@@ -52,10 +52,10 @@ on:
         description: "Container image for the shell check job. Defaults to latest Swift Ubuntu image."
         default: "swift:5.10-noble"
 
-## We are cancelling previously triggered workflow runs
+## Cancel previous runs for the same PR, unless the workflow is running on the main branch, then don't cancel it so we have historical status checks.
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}-soundness
-  cancel-in-progress: true
+  cancel-in-progress: ${{ github.repository.default_branch != github.ref_name }}
```

Comment:
One thing that I came across recently is that we can define the `concurrency` on the call of a reusable workflow. I think we should explore doing that instead of defining it here. This would allow us to just not configure it for the `main` and `scheduled` workflows.

Reply:
Do you mean in the calling repo's main workflow yaml files? That's where I was assuming they'd go — I was actually surprised to see them defined here. If we prefer that approach, I can move these to `pull-request.yml` and `pull-request-label.yml`, and let other repos that depend on these do the same.

Reply:
Not as a top level `concurrency` definition since that didn't work but as a job level `concurrency`. Would be interested to try that out

Reply:
Curious to hear why a top level concurrency definition didn't work (at least in a file like `pull-request.yml`) — that's always been where I've put mine. Happy to try either though, it shouldn't matter too much.

Reply:
(Also, this brings up the question of if we need to pass a `name` to all of these, since it's no longer being used for this purpose (I'll double check the rest of the workflows to double check)

Reply:
The top level concurrency definitions do work but it makes it more complicated since you gotta pass in workflow inputs for these reusable workflows that influence the concurrency group names. We can probably use GitHub context here instead but I was interested to see if it just easier if we define them when calling a reusable workflow. 
It might be that it's not. I just wanted to see it tried out.

Reply:
I guess I meant top-level as in the workflow file that calls all the other workflow files (ie. on pull-request). I'd like to propose this, actually, as it makes it more obvious that concurrency cancellation is in fact enabled, there is no need to pass extra parameters to inform the groups, and it becomes defined in a single place. Let me push a version that shows this.

Only downside is every repo will need to copy the three lines that describe the concurrency group, but as mentioned, I think this clarifies things more than anything. What do you think?

Reply:
https://github.com/apple/swift-nio/commit/ff8843cdaff4ee73cbe3615cee2de3c90091a57b